### ISSUE TITLE
Unicorn doesn't use `CSRF_COOKIE_NAME` from settings

### ISSUE THREAD

**Issue by frnidito** [adamghill/django-unicorn#544]
In https://github.com/adamghill/django-unicorn/blob/main/django_unicorn/static/unicorn/js/utils.js#L52 the function `getCsrfToken` uses a hardcoded value for the token name to read from the header but, according to django documentation, such value can be changes through a setting value making the code not picking-up the right value for the header

```
/**
 * Get the CSRF token used by Django.
 */
export function getCsrfToken(component) {
  // Default to looking for the CSRF in the cookie
  const cookieKey = "csrftoken=";  // <-- *HERE*
  const csrfTokenCookie = component.document.cookie
    .split(";")
    .filter((item) => item.trim().startsWith(cookieKey));

  if (csrfTokenCookie.length > 0) {
    return csrfTokenCookie[0].replace(cookieKey, "");
  }

  // Fall back to check for the CSRF hidden input
  const csrfElements = component.document.getElementsByName(
    "csrfmiddlewaretoken"
  );

  if (csrfElements && csrfElements.length > 0) {
    return csrfElements[0].getAttribute("value");
  }

  throw Error("CSRF token is missing. Do you need to add {% csrf_token %}?");
}
```

In the same way as we can pass to the `Unicorn.ini(...)` the value for the `CSRF_TOKEN_NAME`, we should pass and use the value in `CSRF_COOKIE_NAME`

**Comment by frnidito** [comment]
In the meantime, I wrote and tried locally, a small patch that should fix the issue

```
diff --git a/django_unicorn/static/unicorn/js/component.js b/django_unicorn/static/unicorn/js/component.js
index 9f0cd5f..fc71ed6 100644
--- a/django_unicorn/static/unicorn/js/component.js
+++ b/django_unicorn/static/unicorn/js/component.js
@@ -26,6 +26,7 @@ export class Component {
     this.key = args.key;
     this.messageUrl = args.messageUrl;
     this.csrfTokenHeaderName = args.csrfTokenHeaderName;
+    this.csrfTokenCookieName = args.csrfTokenCookieName;
     this.reloadScriptElements = args.reloadScriptElements;
     this.hash = args.hash;
     this.data = args.data || {};
diff --git a/django_unicorn/static/unicorn/js/unicorn.js b/django_unicorn/static/unicorn/js/unicorn.js
index 18211cf..4b67ffa 100644
--- a/django_unicorn/static/unicorn/js/unicorn.js
+++ b/django_unicorn/static/unicorn/js/unicorn.js
@@ -5,11 +5,12 @@ import { components, lifecycleEvents } from "./store.js";
 let messageUrl = "";
 let reloadScriptElements = false;
 let csrfTokenHeaderName = "X-CSRFToken";
+let csrfTokenCookieName = "csrftoken"

 /**
  * Initializes the Unicorn object.
  */
-export function init(_messageUrl, _csrfTokenHeaderName, _reloadScriptElements) {
+export function init(_messageUrl, _csrfTokenHeaderName, _csrfTokenCookieName, _reloadScriptElements) {
   messageUrl = _messageUrl;
   reloadScriptElements = _reloadScriptElements || false;

@@ -17,9 +18,14 @@ export function init(_messageUrl, _csrfTokenHeaderName, _reloadScriptElements) {
     csrfTokenHeaderName = _csrfTokenHeaderName;
   }

+  if (hasValue(_csrfTokenCookieName)) {
+    csrfTokenCookieName = _csrfTokenCookieName;
+  }
+
   return {
     messageUrl,
     csrfTokenHeaderName,
+    csrfTokenCookieName,
     reloadScriptElements,
   };
 }
@@ -30,6 +36,7 @@ export function init(_messageUrl, _csrfTokenHeaderName, _reloadScriptElements) {
 export function componentInit(args) {
   args.messageUrl = messageUrl;
   args.csrfTokenHeaderName = csrfTokenHeaderName;
+  args.csrfTokenCookieName = csrfTokenCookieName;
   args.reloadScriptElements = reloadScriptElements;

   const component = new Component(args);
diff --git a/django_unicorn/static/unicorn/js/utils.js b/django_unicorn/static/unicorn/js/utils.js
index 5e40553..fe8b072 100644
--- a/django_unicorn/static/unicorn/js/utils.js
+++ b/django_unicorn/static/unicorn/js/utils.js
@@ -51,7 +51,7 @@ export function $(selector, scope) {
  */
 export function getCsrfToken(component) {
   // Default to looking for the CSRF in the cookie
-  const cookieKey = "csrftoken=";
+  const cookieKey = component.csrfTokenCookieName + "="
   const csrfTokenCookie = component.document.cookie
     .split(";")
     .filter((item) => item.trim().startsWith(cookieKey));
diff --git a/django_unicorn/templates/unicorn/scripts.html b/django_unicorn/templates/unicorn/scripts.html
index 03e183f..33ed630 100644
--- a/django_unicorn/templates/unicorn/scripts.html
+++ b/django_unicorn/templates/unicorn/scripts.html
@@ -11,7 +11,7 @@
     reloadScriptElements = true;
   }

-  Unicorn.init(url, "{{ CSRF_HEADER_NAME }}", reloadScriptElements);
+  Unicorn.init(url, "{{ CSRF_HEADER_NAME }}", "{{ CSRF_COOKIE_NAME }}"," reloadScriptElements);
 </script>
 {% else %}
 <script type="module">
@@ -27,6 +27,6 @@
       reloadScriptElements = true;
     }

-    Unicorn.init(url, "{{ CSRF_HEADER_NAME }}", reloadScriptElements);
+    Unicorn.init(url, "{{ CSRF_HEADER_NAME }}", "{{ CSRF_COOKIE_NAME }}", reloadScriptElements);
 </script>
 {% endif %}
\ No newline at end of file
diff --git a/django_unicorn/templatetags/unicorn.py b/django_unicorn/templatetags/unicorn.py
index 728f68a..3fd8753 100644
--- a/django_unicorn/templatetags/unicorn.py
+++ b/django_unicorn/templatetags/unicorn.py
@@ -25,9 +25,12 @@ def unicorn_scripts():

     csrf_header_name = csrf_header_name.replace("_", "-")

+    csrf_cookie_name = settings.CSRF_COOKIE_NAME
+
     return {
         "MINIFIED": get_setting("MINIFIED", not settings.DEBUG),
         "CSRF_HEADER_NAME": csrf_header_name,
+        "CSRF_COOKIE_NAME": csrf_cookie_name,
         "RELOAD_SCRIPT_ELEMENTS": get_setting("RELOAD_SCRIPT_ELEMENTS", False),
     }

diff --git a/tests/js/unicorn/init.test.js b/tests/js/unicorn/init.test.js
index e3cae1b..96b25af 100644
--- a/tests/js/unicorn/init.test.js
+++ b/tests/js/unicorn/init.test.js
@@ -2,25 +2,28 @@ import test from "ava";
 import { init } from "../../../django_unicorn/static/unicorn/js/unicorn.js";

 test("init unicorn", (t) => {
-  const actual = init("unicorn/", "X-Unicorn");
+  const actual = init("unicorn/", "X-Unicorn", "unicorn");

   t.true(actual.messageUrl === "unicorn/");
   t.true(actual.csrfTokenHeaderName === "X-Unicorn");
+  t.true(actual.csrfTokenCookieName === "unicorn");
   t.false(actual.reloadScriptElements);
 });

 test("init unicorn with no reload", (t) => {
-  const actual = init("unicorn/", "X-Unicorn", false);
+  const actual = init("unicorn/", "X-Unicorn", "unicorn", false);

   t.true(actual.messageUrl === "unicorn/");
   t.true(actual.csrfTokenHeaderName === "X-Unicorn");
+  t.true(actual.csrfTokenCookieName === "unicorn");
   t.false(actual.reloadScriptElements);
 });

 test("init unicorn with reload", (t) => {
-  const actual = init("unicorn/", "X-Unicorn", true);
+  const actual = init("unicorn/", "X-Unicorn", "unicorn", true);

   t.true(actual.messageUrl === "unicorn/");
   t.true(actual.csrfTokenHeaderName === "X-Unicorn");
+  t.true(actual.csrfTokenCookieName === "unicorn");
   t.true(actual.reloadScriptElements);
 });
diff --git a/tests/templatetags/test_unicorn_scripts.py b/tests/templatetags/test_unicorn_scripts.py
index 3e5ab5d..9cb5bb7 100644
--- a/tests/templatetags/test_unicorn_scripts.py
+++ b/tests/templatetags/test_unicorn_scripts.py
@@ -5,6 +5,7 @@ def test_unicorn_scripts():
     actual = unicorn_scripts()

     assert actual["CSRF_HEADER_NAME"] == "X-CSRFTOKEN"
+    assert actual["CSRF_COOKIE_NAME"] == "csrftoken"
     assert actual["MINIFIED"] is True


@@ -13,6 +14,7 @@ def test_unicorn_scripts_debug(settings):
     actual = unicorn_scripts()

     assert actual["CSRF_HEADER_NAME"] == "X-CSRFTOKEN"
+    assert actual["CSRF_COOKIE_NAME"] == "csrftoken"
     assert actual["MINIFIED"] is False


@@ -21,6 +23,7 @@ def test_unicorn_scripts_minified_true(settings):
     actual = unicorn_scripts()

     assert actual["CSRF_HEADER_NAME"] == "X-CSRFTOKEN"
+    assert actual["CSRF_COOKIE_NAME"] == "csrftoken"
     assert actual["MINIFIED"] is True


@@ -36,3 +39,9 @@ def test_unicorn_scripts_csrf_header_name(settings):
     actual = unicorn_scripts()

     assert actual["CSRF_HEADER_NAME"] == "X-UNICORN"
+
+def test_unicorn_scripts_csrf_cookie_name(settings):
+    settings.CSRF_COOKIE_NAME = "unicorn-csrftoken"
+    actual = unicorn_scripts()
+
+    assert actual["CSRF_COOKIE_NAME"] == "unicorn-csrftoken"
```

It locally passes all the tests, old and new, for both py and js

**Comment by adamghill** [comment]
Makes sense that we should use that setting. If you want to make a PR I can merge it in a few weeks (currently without a laptop). Otherwise, I can use your patch and credit you manually in the changelog.

**Comment by frnidito** [comment]
I don't have permission to create PRs so you can either enable me and I can add the PR myself or use directly my patch if that is ok for you. I have no preferences on the matter :)

**Comment by adamghill** [comment]
Everyone should be able to fork the repo and make a PR, but if that doesn't work then I can do it when I get a chance!

**Comment by frnidito** [comment]
Sorry, my bad, I wanted to create a PR starting from the main repo and from its fork. I opened it now.

Just a quick question: do I need to rebuild and add to the PR the minified too or is that done only when a new release goes out?

**Comment by adamghill** [comment]
No worries! Thanks for the PR, I'll take a look when I get access to a laptop. You don't need to minify, I'll do that for the release. 👍

**Comment by adamghill** [comment]
#545 merged in and will be included in v0.52.0. Thanks for all the help! 🚀
